### PR TITLE
Add list of marker poops

### PR DESCRIPTION
**Pull request by antoniosj**

### CODE REVIEW

File: app/src/main/java/com/discordtime/gpts/maps/viewmodel/MapViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import com.discordtime.gpts.repository.PlaceRepository
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.BitmapDescriptor
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+
+class MapViewModel: ViewModel() {
+
+    private val repository: IPlaceRepository = PlaceRepository()
```
Comment (viniciusalbuquerque) [inline app/src/main/java/com/discordtime/gpts/maps/viewmodel/MapViewModel.kt:21]: Why don't we inject this? You defined a module for PlaceRepository, consider using it.

**Review by carlosolimpio — APPROVED** [review]

File: app/src/main/java/com/discordtime/gpts/remote/FirestoreAPI.kt [hunk 2]
```diff
@@ -2,11 +2,32 @@ package com.discordtime.gpts.remote
 
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
+import com.discordtime.gpts.listplaces.model.Place
+import com.discordtime.gpts.tools.Constants.PLACES_COLLECTION
+import com.google.firebase.firestore.CollectionReference
+import com.google.firebase.firestore.FirebaseFirestore
 
-class FirestoreAPI<T>: RemoteAPI<T> {
+class FirestoreAPI: RemoteAPI<Place> {
```
Comment (viniciusalbuquerque) [inline app/src/main/java/com/discordtime/gpts/remote/FirestoreAPI.kt:10]: I think we should not change this from generics for Place. I think we should leave it as T, return the snapshot and have a mapper that would transform the snapshot on the entity we want.

File: app/src/main/java/com/discordtime/gpts/repository/PlaceRepository.kt [hunk 3]
```diff
@@ -2,9 +2,12 @@ package com.discordtime.gpts.repository
...
+import com.discordtime.gpts.remote.FirestoreAPI
 import com.discordtime.gpts.remote.RemoteAPI
 
-class PlaceRepository(val remoteAPI: RemoteAPI<Place>): IPlaceRepository {
+class PlaceRepository: IPlaceRepository {
+
+    private val remoteAPI: RemoteAPI<Place> = FirestoreAPI()
 
     override fun getPlaces(): LiveData<List<Place>> {
         return remoteAPI.get()
```
Comment (viniciusalbuquerque) [inline app/src/main/java/com/discordtime/gpts/repository/PlaceRepository.kt:13]: As I mentioned above, I think we could have a mapper here that would take care of the transformation from what is returned from the remote api. I think we could put it the mapper on the constructor and have it to be injected on the Koin module.

File: app/src/main/java/com/discordtime/gpts/maps/view/MapViewFragment.kt [hunk 4]
```diff
@@ -28,43 +36,33 @@ class MapViewFragment : Fragment(), OnMapReadyCallback {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
         mMapView = view.findViewById(R.id.mapView) as MapView
         mMapView.onCreate(savedInstanceState)
         mMapView.onResume()
         mMapView.getMapAsync(this)
+        mapViewModel = ViewModelProviders.of(this)[MapViewModel::class.java]
```
Comment (viniciusalbuquerque) [inline app/src/main/java/com/discordtime/gpts/maps/view/MapViewFragment.kt:44]: A question, when we're using Koin, should we use the "by inject" that it provides or we should use the ViewModelProviders from the Architeture Components.

Maybe if we use Koin we can start to inject what we need there on the ViewModel (e.g. IPlaceRepository).

**Review by viniciusalbuquerque — CHANGES_REQUESTED** [review]

**Pull request merged**